File: diceroller.py
from random import randint


def roll_xdy(numberToRoll, dieSides):
    """
    Returns a generator of dice roll results.

    Parameters:
        numberToRoll: Integer
            The number of dice to roll.
        dieSides: Integer
            The number of sides the dice being rolled have.
    """
    total = 0
    for _ in range(numberToRoll):
        total += randint(1, dieSides)
    return total

File: scifi_gen/src/dice_roller.rs
use rand::{Rng, RngExt};

pub fn roll_xdy<R: Rng>(rng: &mut R, number_to_roll: u8, die_sides: u8) -> u8 {
    let mut total: u8 = 0;
    for _ in 0..number_to_roll {
        total += rng.random_range(1..=die_sides);
    }
    total
}